test/constant.cpp

```diff
@@ -49,7 +49,7 @@ int main()
         static_assert(
             std::is_same<
                 decltype(1844674407370955161_c),
-                smath::constant<long long, 1844674407370955161>
+                smath::constant<long, 1844674407370955161>
```

Comment:
I probably doesn't work with Visual Studio at all since its `constexpr` support is wonky. I use MinGW-w64.